src/lib.rs

```diff
@@ -238,16 +238,27 @@ impl DepRefresh {
 
             for (the_crate, local_version, online_version) in updates_crate {
                 self.update_info(&the_crate, &local_version, &online_version);
-                doc[table_name][&the_crate] = toml_edit::value(online_version);
+                let new_local_version = self.new_local_version(&local_version, &online_version);
+                doc[table_name][&the_crate] = toml_edit::value(new_local_version);
             }
             for (the_crate, local_version, online_version) in updates_crate_version {
                 self.update_info(&the_crate, &local_version, &online_version);
-                doc[table_name][&the_crate]["version"] = toml_edit::value(online_version);
+                let new_local_version = self.new_local_version(&local_version, &online_version);
+                doc[table_name][&the_crate]["version"] = toml_edit::value(new_local_version);
             }
         }
         Ok(())
     }
 
+    // generate new version format with same segaments as the local_version
+    fn new_local_version<'a>(&self, local_version: &str, online_version: &'a str) -> &'a str {
+        let mut new_local_version = online_version;
+        for _ in 0..(3 - local_version.split('.').collect::<Vec<&str>>().len()) {
+            new_local_version = new_local_version.rsplit_once('.').map(|(head, _tail)| head).unwrap_or(online_version);
+        }
```

Comment:
Does this work with beta and prerelease versions?
We do already use semver::VersionReq in another part of the crate to parse version strings.
I'd say it would be better to use that here, too, instead of spinning our own version string parser for local_version and online_version.